cardano-cli/src/Cardano/CLI/Run/Debug/Transaction/Echo.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.CLI.Run.Debug.Transaction.Echo
+  ( runDebugTransactionEchoCmd
+  ) where
+
+import           Cardano.Api
+
+import qualified Cardano.CLI.Commands.Debug.Transaction.Echo as Cmd
+import           Cardano.CLI.Orphans ()
+import           Cardano.CLI.Read
+import           Cardano.CLI.Types.Common
+import           Cardano.CLI.Types.Errors.DebugCmdError
+
+import           Data.Function ((&))
+
+runDebugTransactionEchoCmd :: ()
+  => Cmd.TransactionEchoCmdArgs
+  -> ExceptT DebugCmdError IO ()
+runDebugTransactionEchoCmd
+    Cmd.TransactionEchoCmdArgs
+      { Cmd.txOrTxBodyFile = txOrTxBody
+      , Cmd.outTxFile = outTxFile
+      } = do
+
+  case txOrTxBody of
+    InputTxFile (File inputTxFilePath) -> do
+      inputTxFile <- liftIO $ fileOrPipe inputTxFilePath
+      anyTx <- lift (readFileTx inputTxFile) & onLeft (left . DebugCmdTextEnvCddlError)
+
+      InAnyShelleyBasedEra sbe tx <- pure anyTx
```

Comment:
```suggestion
      InAnyShelleyBasedEra sbe tx <- lift (readFileTx inputTxFile) & onLeft (left . DebugCmdTextEnvCddlError)
```

I was thinking that this would be equivalent to:
```haskell
let InAnyShelleyBasedEra sbe tx = anyTx
```
But it is not. The `let` version doesn't compile, interesting 🤔
But still can do the suggestion, unless you are doing it for clarity.